### PR TITLE
SDD 0028 - Reusable Commodore Component Configuration Packages

### PR DESCRIPTION
**Pull request by simu**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.

### CODE REVIEW

**Review by glrf — APPROVED** [review]
This looks like a good approach.

As for the question of using remote inventories or implementing a custom solution: If I understand the approach correctly, using Kapitan remote inventories will lead to package fetching working differently than a user would expect (e.g. not being able to overwrite already included packages).

I think I would slightly prefer the custom approach. This will lead to a bit more custom code, but probably leads to a better user experience.

File: docs/modules/SDDs/pages/0028-reusable-config-packages.adoc [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+* Programmatic manipulation of component defaults
+
+== Design Proposal
+
+We propose that each Commodore component configuration package is a Git repository.
+This gives us change history and versioning of configuration packages based on a well-established toolset.
+Making each configuration package a Git repository also fits well with the Project Syn approach of keeping all dependencies (global and tenant config, components) in Git.
+We can also profit from all the already existing logic for cutting releases, and CI tests which we've engineered for Commodore components.
+
+The big design question is revolves around the mechanism used to fetch configuration packages.
```
Comment (corvus-ch) [inline docs/modules/SDDs/pages/0028-reusable-config-packages.adoc]:
```suggestion
The big design question revolves around the mechanism used to fetch configuration packages.
```

**Review by corvus-ch — APPROVED** [review]

**Comment by simu** [comment]
> This looks like a good approach.
> 
> As for the question of using remote inventories or implementing a custom solution: If I understand the approach correctly, using Kapitan remote inventories will lead to package fetching working differently than a user would expect (e.g. not being able to overwrite already included packages).
> 
> I think I would slightly prefer the custom approach. This will lead to a bit more custom code, but probably leads to a better user experience.

Please see the latest version. I've now defined the package specification and inclusion more explicitly. This changes the trade-off between Kapitan's remote inventory and Commodore-native remote inventory a bit, as we now have to transform the user-provided config into Kapitan's format if we want to use Kapitan's remote inventory. My guess is that this will reduce the difference in implementation complexity between the native and Kapitan fetching from the earlier proposal.

File: docs/modules/SDDs/pages/0028-reusable-config-packages.adoc [hunk 2]
```diff
@@ -45,33 +45,49 @@ This gives us change history and versioning of configuration packages based on a
 Making each configuration package a Git repository also fits well with the Project Syn approach of keeping all dependencies (global and tenant config, components) in Git.
 We can also profit from all the already existing logic for cutting releases, and CI tests which we've engineered for Commodore components.
 
+=== Specifying configuration packages
+
+We propose a new top-level parameter `configuration_packages`.
```
Comment (glrf) [inline docs/modules/SDDs/pages/0028-reusable-config-packages.adoc]: Why not just `packages`?

**Comment by glrf** [comment]
> Please see the latest version. I've now defined the package specification and inclusion more explicitly. This changes the trade-off between Kapitan's remote inventory and Commodore-native remote inventory a bit, as we now have to transform the user-provided config into Kapitan's format if we want to use Kapitan's remote inventory. My guess is that this will reduce the difference in implementation complexity between the native and Kapitan fetching from the earlier proposal

Looks good, this is more or less how I expected the custom approach to work. You're right we can probably make it work the same way when using Kapitan's remote inventory. It just feels a bit "hacky"? I would probably still go with the custom approach, but I don't really have strong feelings either way

File: docs/modules/SDDs/pages/0028-reusable-config-packages.adoc (hunk 2 above)
Comment (simu) [reply docs/modules/SDDs/pages/0028-reusable-config-packages.adoc]: I've no hard feelings either way, I'd be fine with using `packages`.

Comment (simu) [reply docs/modules/SDDs/pages/0028-reusable-config-packages.adoc]: Changed to `packages` in the latest version

**Comment by simu** [comment]
> > Please see the latest version. I've now defined the package specification and inclusion more explicitly. This changes the trade-off between Kapitan's remote inventory and Commodore-native remote inventory a bit, as we now have to transform the user-provided config into Kapitan's format if we want to use Kapitan's remote inventory. My guess is that this will reduce the difference in implementation complexity between the native and Kapitan fetching from the earlier proposal
> 
> Looks good, this is more or less how I expected the custom approach to work. You're right we can probably make it work the same way when using Kapitan's remote inventory. It just feels a bit "hacky"? I would probably still go with the custom approach, but I don't really have strong feelings either way

I'm tending towards the Commodore-native implementation as well, since I think we'll end up with a cleaner implementation, even if there's a bit more of it.

**Pull request merged**